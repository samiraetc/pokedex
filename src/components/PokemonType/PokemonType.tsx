import { Pokemon } from "../../types/typing";
import { pokemonTypeColors } from "../../utils/pokemon-type-colors";

export const COMPONENT_ID = 'pokemon-details-type';

export type PokemonTypeProps = {
    type: string,
}

const PokemonType = ({type}: PokemonTypeProps) => {
    return <p id={COMPONENT_ID} className={`${COMPONENT_ID}__item`} style={{ backgroundColor: pokemonTypeColors[type] }}>{type}</p>
}

export default PokemonType;