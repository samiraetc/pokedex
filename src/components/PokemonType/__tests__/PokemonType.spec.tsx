import { render, configure } from '@testing-library/react'
import PokemonType, { COMPONENT_ID } from '../PokemonType';

configure({
    testIdAttribute: 'id',
  });
  
describe('[POKEMONTYPE]', () => {
    test('should render PokemonType', () => {
        const { asFragment, getByText, getByTestId } = render(<PokemonType type={'normal'} />);

        expect(getByText('normal'));
        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
});