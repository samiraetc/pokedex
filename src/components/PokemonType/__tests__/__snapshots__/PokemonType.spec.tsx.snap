// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[POKEMONTYPE] should render PokemonType 1`] = `
<DocumentFragment>
  <p
    class="pokemon-details-type__item"
    id="pokemon-details-type"
    style="background-color: rgb(168, 167, 122);"
  >
    normal
  </p>
</DocumentFragment>
`;
